arangod/RestServer/ViewTypesFeature.cpp

```diff
@@ -27,43 +27,51 @@
 #include "ProgramOptions/Section.h"
 #include "Views/LoggerView.h"
 
-using namespace arangodb;
-using namespace arangodb::application_features;
-using namespace arangodb::basics;
-using namespace arangodb::options;
+namespace {
 
-std::unordered_map<std::string, arangodb::ViewCreator>
-    ViewTypesFeature::_viewCreators;
+std::string const FEATURE_NAME("ViewTypes");
+arangodb::ViewCreator const INVALID{};
 
-ViewTypesFeature::ViewTypesFeature(ApplicationServer* server)
-    : ApplicationFeature(server, "ViewTypes") {
+} // namespace
+
+namespace arangodb {
+
+ViewTypesFeature::ViewTypesFeature(
+  application_features::ApplicationServer* server
+): application_features::ApplicationFeature(server, ViewTypesFeature::name()) {
   setOptional(false);
   requiresElevatedPrivileges(false);
   startsAfter("WorkMonitor");
 }
 
-void ViewTypesFeature::prepare() {
-  // register the "logger" example view type
-  registerViewImplementation(LoggerView::type, LoggerView::creator);
+bool ViewTypesFeature::emplace(
+    LogicalDataSource::Type const& type,
+    ViewCreator creator
+) {
+  return _factories.emplace(&type, creator).second;
 }
 
-void ViewTypesFeature::unprepare() { _viewCreators.clear(); }
+ViewCreator const& ViewTypesFeature::factory(
+    LogicalDataSource::Type const& type
+) const noexcept {
+  auto itr = _factories.find(&type);
 
-void ViewTypesFeature::registerViewImplementation(std::string const& type,
-                                                  ViewCreator creator) {
-  _viewCreators.emplace(type, creator);
+  return itr == _factories.end() ? INVALID : itr->second;
 }
 
-bool ViewTypesFeature::isValidType(std::string const& type) const {
-  return (_viewCreators.find(type) != _viewCreators.end());
+/*static*/ std::string const& ViewTypesFeature::name() {
+  return FEATURE_NAME;
 }
 
-ViewCreator& ViewTypesFeature::creator(std::string const& type) const {
-  auto it = _viewCreators.find(type);
-
-  if (it == _viewCreators.end()) {
-    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
-                                   "no handler found for view type");
-  }
-  return (*it).second;
+void ViewTypesFeature::prepare() {
+  // register the "logger" example view type
+  emplace(LoggerView::type(), LoggerView::creator);
```

Comment:
this is outside the scope of the task #344

Reply:
@vasiliy-arangodb The fact of presence LoggerView requires a useless changes in LoggerView.h and LoggerView.cpp to compile ArangoDB successfully which means that it's in the scope of the task. Since LoggerView is no longer needed please remove LoggerView.* files instead of changing them.

Reply:
this is outside the scope of this pull-request